shell/browser/ui/file_dialog_gtk.cc

```diff
@@ -253,11 +253,18 @@ void FileChooserDialog::AddFilters(const Filters& filters) {
 
     for (const auto& extension : filter.second) {
       // guarantee a pure lowercase variant
-      std::string file_extension = base::ToLowerASCII("*." + extension);
-      gtk_file_filter_add_pattern(gtk_filter, file_extension.c_str());
+      std::string lower_file_extension = base::ToLowerASCII("*." + extension);
+      gtk_file_filter_add_pattern(gtk_filter, lower_file_extension.c_str());
+
       // guarantee a pure uppercase variant
-      file_extension = base::ToUpperASCII("*." + extension);
-      gtk_file_filter_add_pattern(gtk_filter, file_extension.c_str());
+      std::string upper_file_extension = base::ToUpperASCII("*." + extension);
+      gtk_file_filter_add_pattern(gtk_filter, upper_file_extension.c_str());
+
+      // if we received a mixed-case extension, add that extension as-is
```

Comment:
If the filter is set to `Foo`, it would still not catch files with the extension `fOO`. Is that expected or should we convert the extension to `*.[fF][oO][oO]` instead in which case it would catch all such files regardless of the casing?

Reply:
That seems reasonable to me. Should I adjust my PR to that effect?

Reply:
Thanks! I've updated the PR following the aforementioned implementation.